### DISCUSSION TITLE
httpx gets stuck - No output

### DISCUSSION THREAD

**Discussion by ta1bbty** [projectdiscovery/httpx#541] [Q&A] [answered]
### httpx version:
`v1.1.5`

### Current Behavior:
I am running this `./naabu -retries 1 -host 13.58.0.0/15 -p 80,443 -rate 50000 -c 500 -silent | ./httpx -ip -cname -cdn -location -title -cl -td -server -sc -t 500 -tls-grab -csp-probe -pipeline -http2 -vhost -json -o 1645762606541-output-13.58.0.0.json`

The resource utilitsation on the server is zero.

<img width="1651" alt="image" src="https://user-images.githubusercontent.com/45117337/155698244-4bfa17d8-caec-44f2-b18c-02e01965b8a8.png">

Even the output from `httpx` is stuck, no more output. I checked the open file descriptor.
```bash
[root@ec2-user]# ulimit -Hn
65535
[root@ec2-user]# ulimit -Sn
65535
[root@ec2-user]# cat /proc/sys/fs/file-max
801198
```

The output log is stuck here :
```bash
[ERR] could not discard response body: read tcp 172.31.1.71:35494->13.59.121.63:80: read: connection timed out (Client.Timeout or context cancellation while reading body)
{"timestamp":"2022-02-25T09:02:13.719050022Z","scheme":"http","port":"80","path":"/","body-sha256":"b0ee315f4ac6af09d05f9e6f23ffb606f3b4fec1ba897bd4315592d2a2979876","header-sha256":"041d0fe8ee6ea2168e566f8d4ef32d7f1bd132fb93ada33a864b14125a629eb4","url":"http://13.59.121.63:80","input":"13.59.121.63:80","location":"https://13.59.121.63:443/","content-type":"text/plain","method":"GET","host":"13.59.121.63","content-length":5,"status-code":302,"pipeline":true,"response-time":"2.258141ms","failed":false,"lines":1,"words":1}
```

### Expected Behavior:
It should've been showing output to `STDOUT`.

### Steps To Reproduce:
1. Run `./naabu -retries 1 -host 13.58.0.0/15 -p 80,443 -rate 50000 -c 500 -silent | ./httpx -ip -cname -cdn -location -title -cl -td -server -sc -t 500 -tls-grab -csp-probe -pipeline -http2 -vhost -json -o 1645762606541-output-13.58.0.0.json`

**Comment by ehsandeep** [comment, marked as answer]
@ta1bbty v1.2.0 is the [latest](https://github.com/projectdiscovery/httpx/releases/tag/v1.2.0) version of the project, make sure to update it to latest to confirm the issue again.
